### PR TITLE
Feat: Remove the framebuffer hub. Manually update framebuffer after attachment or swapchain changes.

### PR DESCRIPTION
**Pull request by bluesky013**
Re: #

### Changelog

* Remove the framebuffer hub. Manually update framebuffer after attachment changes.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by bluesky013** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

File: native/cocos/scene/RenderWindow.cpp [hunk 1]
```diff
@@ -143,6 +139,34 @@ void RenderWindow::extractRenderCameras(ccstd::vector<Camera *> &cameras) {
...
+        return;
+    }
+    _swapchain->createSurface(hWnd);
+    updateFramebuffer();
+}
+
+void RenderWindow::updateFramebuffer() {
+    if (_frameBuffer != nullptr) {
+        _frameBuffer->destroy();
+        _frameBuffer->initialize({
```
Comment (hana-alice) [inline native/cocos/scene/RenderWindow.cpp]: we got a convention that GFXObject should not be reinit. 
way of:
```
_frameBuffer->destroy();
_frameBuffer->initialize(info);
...
```
may be changed to
```
_frameBuffer->destroy(); // do not reuse
_frameBuffer = device->createFramebuffer(info);
...
```

Comment (minggo) [reply native/cocos/scene/RenderWindow.cpp]: In cpp, should delete, not just destroy if want to create a new one. And i agree should not reuse gfx object.

Comment (bluesky013) [reply native/cocos/scene/RenderWindow.cpp]: I agree with that. Gfx objects should not initialize twice.

File: native/cocos/scene/RenderWindow.h [hunk 2]
```diff
@@ -65,8 +65,8 @@ class RenderWindow : public RefCounted {
 
     void extractRenderCameras(ccstd::vector<Camera *> &cameras);
 
-    void onNativeWindowDestroy(uint32_t windowId);
-    void onNativeWindowResume(uint32_t windowId);
+    void onNativeWindowDestroy(void *windowId);
+    void onNativeWindowResume(void *windowId);
```
Comment (minggo) [inline native/cocos/scene/RenderWindow.h]: Why use void*?

Comment (bluesky013) [reply native/cocos/scene/RenderWindow.h]: I use ```void*``` to match args from ```EventDispatcher```. It seems a little weird DESTROY_EVENT use windowHandle while RESUME_EVENT use windowId.

Comment (minggo) [reply native/cocos/scene/RenderWindow.h]: @freeboy i think should use window id both in resume and destroy event.

File: native/cocos/scene/RenderWindow.cpp [hunk 3]
```diff
@@ -143,6 +136,32 @@ void RenderWindow::extractRenderCameras(ccstd::vector<Camera *> &cameras) {
...
+    auto *windowMgr = BasePlatform::getPlatform()->getInterface<ISystemWindowManager>();
+    auto *hWnd = reinterpret_cast<void *>(windowMgr->getWindow(windowId)->getWindowHandle());
+    _swapchain->createSurface(hWnd);
+    updateFramebuffer();
+}
+
+void RenderWindow::updateFramebuffer() {
+    if (_frameBuffer != nullptr) {
+        _frameBuffer = nullptr;
+    }
```
Comment (minggo) [inline native/cocos/scene/RenderWindow.cpp]: I think it is not needed as IntrusivePtr will handle it when be assigned to a new value.

Comment (minggo) [reply native/cocos/scene/RenderWindow.cpp]: And the function name is better to rename to `generateFrameBuffer` or similar meaning. update means update the contents of a object.

Comment (bluesky013) [reply native/cocos/scene/RenderWindow.cpp]: Got it.

**Review by minggo — APPROVED** [review]

**Review by star-e — APPROVED** [review]
looks good.

**Comment by sushanhong** [comment]
@cocos-robot run test case

**Pull request merged**